crates/astria-core/src/protocol/transaction/v1alpha1/action.rs

```diff
@@ -45,6 +45,50 @@ pub enum Action {
     FeeChange(FeeChangeAction),
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum PriorityCategory {
+    BundlableGeneral,
+    General,
+    BundleableSudo,
+    Sudo,
+}
+
+pub trait Priority {
+    fn priority(&self) -> PriorityCategory;
+}
+
+pub trait Bundleable {
+    fn is_bundleable(&self) -> bool;
+}
```

Comment:
I'm realizing now we probably don't even need the traits this can just be an impl. No need for the trait if we don't need multiple types implementing it.